agnpy/absorption/absorption.py

```diff
@@ -479,6 +588,67 @@ def tau_blr_mu_s(self, nu):
             phi=self.phi,
         )
 
+    def tau_blr_cubepy(self, nu, eps_abs=1e-6):
+        """Evaluates the gamma-gamma absorption produced by a spherical shell
+        BLR for a general set of model parameters with cubepy integration
+        method
+        """
+
+        tau_blr_list = []
+        for freq in nu:
+            tau_blr_list.append(
+            self.evaluate_tau_blr_cubepy(
+                freq,
+                eps_abs,
+                self.z,
+                self.mu_s,
+                self.target.L_disk,
+                self.target.xi_line,
+                self.target.epsilon_line,
+                self.target.R_line,
+                self.r
+            )
+            )
+        return tau_blr_list
+
+    def tau_blr_all_lines_cubepy(self, nu, eps_abs=1e-6):
+        """
+        Calculate the absorption in all available BLR (Broad Line Region) shells.
+        This function scales the radius and luminosity to the Hbeta line
+        and applies the computation to all the available BLR lines (shells) using
+        their respective ratios.
+
+        Parameters
```

Comment:
how about adding a dictionary 'lines' to the parameters (and setting it by default to lines_dictionary)
this would allow to run the function with limited number of lines (selecting only the most important ones for speed) or with modified dictionary (if we have more accurate measurements for a particular source). It would also facilitate tests because you could then implement a quick test on a dictionary of 2 lines

Reply:
Great idea, but what do you think about doing it in the "second" iteration (after merging this code with all lines into the main branch)?

Reply:
if you prefer ok, but note that this is really minor change to the code, just adding a new parameter to the function, with a default value and then in the line 637 using this parameter instead of lines_dictionary.
And it would simplify implementation of my comment about tests

Reply:
I have created a new function, `tau_blr_lines_cubepy`. Could you look?

Reply:
thanks. I put some comments there, and I am closing the comment here